src/components/Article.jsx

```diff
@@ -1,40 +1,117 @@
 import React, { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 import { fetchArticle } from "../utils/api";
+import { patchVotes } from "../utils/api";
 import Loading from "./Loading";
+import Comments from "./Comments";
 
 const Article = () => {
     const { article_id } = useParams()
     const [article, setArticle] = useState({})
     const [isLoading, setIsLoading] = useState(true)
+    const [voteError, setVoteError] = useState(false)
+    const [votes, setVotes] = useState(0)
+    const [upVoted, setUpVoted] = useState(false)
+    const [downVoted, setDownVoted] = useState(false)
+
 
     useEffect(()=> {
         fetchArticle(article_id)
         .then((response)=> {
             setIsLoading(false)
             setArticle(response)
+            setVotes(response.votes)
         })
     }, [])
 
-    
+    const upVote = () => {
+        if (!upVoted) {
+            setUpVoted(true)
+            setVotes(votes + 1)
+            return patchVotes(article_id, 1)
+            .then((response) => {
+                setVotes(response.votes)
+            })
+            .catch((err) => {
+                console.log(err.response.status)
+                console.log(err.response.data.msg)
+                setVoteError(true)
+            })
+        } else {
+            setUpVoted(false)
+            setVotes(votes -1)
+            return patchVotes(article_id, -1)
+            .then((response) => {
+                setVotes(response.votes)
+            })
+            .catch((err) => {
```

Comment:
Please make sure that in case of an error, the votes displayed are reverted to the original to avoid any confusion. At the moment when the app is offline, it's quite confusing when votes are clicked.